packages/grafana-ui/src/components/TransformersUI/FilterByNameTransformerEditor.tsx

```diff
@@ -97,36 +101,69 @@ export class FilterByNameTransformerEditor extends React.PureComponent<
   };
 
   onChange = (selected: string[]) => {
+    const { regex, isRegexValid } = this.state;
+    let include = selected;
+
+    if (regex && isRegexValid) {
+      include = include.concat([regex]);
+    }
+
     this.setState({ selected }, () => {
       this.props.onChange({
         ...this.props.options,
-        include: this.state.regex ? [...selected, this.state.regex] : selected,
+        include,
       });
     });
   };
 
   onInputBlur = (e: React.FocusEvent<HTMLInputElement>) => {
     const { selected, regex } = this.state;
-    this.props.onChange({
-      ...this.props.options,
-      include: regex ? [...selected, regex] : selected,
+    let isRegexValid = true;
+    try {
+      if (regex) {
+        new RegExp(regex);
+      }
+    } catch (e) {
+      isRegexValid = false;
+    }
+    if (isRegexValid) {
+      this.props.onChange({
+        ...this.props.options,
+        include: regex ? [...selected, regex] : selected,
+      });
+    } else {
+      this.props.onChange({
+        ...this.props.options,
+        include: selected,
+      });
+    }
+    this.setState({
+      isRegexValid,
     });
   };
 
   render() {
-    const { options, selected } = this.state;
+    const { options, selected, isRegexValid } = this.state;
     return (
       <div className="gf-form-inline">
         <div className="gf-form gf-form--grow">
           <div className="gf-form-label width-8">Field name</div>
-          <HorizontalGroup spacing="xs">
-            <Input
-              placeholder="Regular expression pattern"
-              value={this.state.regex || ''}
-              onChange={e => this.setState({ regex: e.currentTarget.value })}
-              onBlur={this.onInputBlur}
-              width={25}
-            />
+          <HorizontalGroup spacing="xs" align="flex-start" wrap>
```

Comment:
@torkelo left this instead gf-form-label for now, I want to tackle this when we improve inline form styles. I like the way how easy it is to get the layout I want with this approach and gonna refactor this as soon as we redesign inline forms.